<a href="https://colab.research.google.com/github/vitorjensen/Python_para_Analise_de_Dados/blob/main/Projetos/An%C3%A1lise%20Explorat%C3%B3ria%20de%20Combust%C3%ADvel/An%C3%A1lise_Explorat%C3%B3ria_de_Combust%C3%ADvel.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Importando bibliotecas necessárias:

In [1]:
import yfinance as yf
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
from datetime import datetime

In [ ]:
# pip install yfinance pandas

**Objetivo:** Fazer uma análise exploratória do Brent (Petróleo bruto) no período de 03/01/2022 a 27/09/2026

**Primeira parte:** Extração da base de dados utilizando o Yahoo Finance.

In [ ]:
# Definindo o ticker (Brent) a ser manipulado
ticker = "BZ=F"

# Criando o dataset com o período de dados estabelecido
data_inicio = "2022-01-01"
data_fim = datetime.now().strftime('%Y-%m-%d')

# Definindo um objeto para o ticker
brent = yf.Ticker(ticker)

# Criando o dataframe com os dados do Yahoo extraídos
df_brent = brent.history(start=data_inicio, end=data_fim)

display(df_brent.head())

In [ ]:
# Fazendo download em CSV da base para a máquina

df_brent = df_brent.to_csv('dados_brent.csv')

**Segunda parte:** Tratamento.

- Remover colunas indesejadas
- Verificaf Total de linhas e colunas
- Renomear colunas
- Verificar ocorrências de valores 0
- Verificar valores duplicados
- Total de valores ausentes
- Verificação do tipo da coluna "Data"
- Verificando Data Início e Data Final
- Verificando Datas Únicas
- Verificando existência de Datas Duplicadas

In [ ]:
# Lendo aquivo CSV
df = pd.read_csv('/content/drive/MyDrive/dados_brent.csv')

# Total de colunas e linhas
display(df)

In [ ]:
# informações gerais do dataset

df.info()

In [ ]:
# Excluindo colunas indesejadas:
colunas_indesejadas = ['Dividends', 'Stock Splits']
df = df.drop(colunas_indesejadas, axis=1, errors='ignore')

display(df)

In [ ]:
# Total de linhas e colunas
print(f'Total de linhas: {df.shape[0]}')
print(f'Total de colunas: {df.shape[1]}')

Total de linhas: 1190
Total de colunas: 6


In [ ]:
# Renomear colunas
df = df.rename(columns={
    'Date': 'Data',
    'Open': 'Valor_Abertura',
    'Low': 'Valor_Minimo',
    'High': 'Valor_Maximo',
    'Close': 'Fechamento'
})

display(df)

In [ ]:
# Verificando existência de valores "zero"

geral_zeros = df[(df == 0).any(axis=1)]
print(f"Quantidade de linhas contendo pelo menos um zero: {len(geral_zeros)}")
display(geral_zeros)

In [ ]:
# Verificando total de duplicados

duplicados = df.duplicated().sum()
print(f'Total de valores duplicados: {duplicados}')

Total de valores duplicados: 0


In [ ]:
# Verificar valores ausentes
print("Valores ausentes por coluna:")
display(df.isnull().sum())

In [ ]:
# Verificando e garantindo o tipo da coluna data
df['Data'] = pd.to_datetime(df['Data'], utc=True)

print(df['Data'].dtype)

datetime64[ns, UTC]


In [ ]:
# Verificando Data Inicial e Data Final

print(f"Data Inicial: {df['Data'].min()}")
print(f"Data Final: {df['Data'].max()}")


Data Inicial: 2022-01-03 05:00:00+00:00
Data Final: 2026-09-25 04:00:00+00:00


In [ ]:
# Verificando datas únicas

print(f"Total de data únicas: {df['Data'].nunique()}")

Total de data únicas: 1190


In [ ]:
# Verificando datas duplicadas

datas_duplicadas = df['Data'].duplicated().sum()
print(f"Quantidade de datas duplicadas: {datas_duplicadas}")

Quantidade de datas duplicadas: 0


In [ ]:
#Ordenando o dataset por data

df = df.sort_values('Data').reset_index(drop=True)
display(df.head())
display(df.tail())

Resultado do tratamento:

- Total de colunas removidas: 2 (Dividends e Stock Split)
- Total de linhas e colunas: 1190 x 6
- Total de colunas renomeadas: 5
- Total de ocorrências de valores 0: 8
- Total de valores duplicados: 0
- Total de valores ausentes: 0
- Verificação do tipo da coluna "Data" = OK
- Verificando Data Início e Data Final = OK
- Verificando Datas Únicas = OK
- Verificando existência de Datas Duplicadas = 0

**Terceira parte:** Análise Exploratória do Brent

- Execução do Describe.
- Colunas categóricas.
- Colunas numéricas e sua Análise Univariada.
- Definir Variação Percentual Diária.
- Correlação Linear
- Outras análises pertinentes ao conjunto de dados.

In [ ]:
# Verificando estatísticas básicas do Brent
df.describe()

In [ ]:
# Verificando colunas categóricas:

colunas_categoricas =[columns for columns in df.columns
                      if df[columns].dtype == 'object']

qtde_total = len(colunas_categoricas)

print(f'Qtde de colunas categóricas: {qtde_total}')

print(f'Colunas categóricas: {colunas_categoricas}')

In [5]:
# Verificando colunas numéricas

colunas_numericas = [columns for columns in df.columns
                     if df[columns].dtype != 'object']

qtde_total = len(colunas_numericas)

print(f'Qtde de colunas numéricas: {qtde_total}')
print(f'Colunas numéricas: {colunas_numericas}')



Qtde de colunas numéricas: 5
Colunas numéricas: ['Valor_Abertura', 'Valor_Maximo', 'Valor_Minimo', 'Fechamento', 'Volume']


In [ ]:
# Definindo as estatísticas baseada nas colunas numéricas

estatistica = df[colunas_numericas].describe().T
display(estatistica)

In [ ]:
# Definindo a Variação Diária em Porcentagem

df['Variacao_Diaria_%'] = df['Fechamento'].pct_change() * 100
display(df)

In [ ]:
# Analisando Variáveis estatísticas da Variação Diária

df['Variacao_Diaria_%'].describe()

In [13]:
# Analisandos os 10 dias com Maiores Altas

maiores_altas = df.nlargest(10, 'Variacao_Diaria_%')
display(maiores_altas)

,Data,Valor_Abertura,Valor_Maximo,Valor_Minimo,Fechamento,Volume,Variacao_Diaria_%
1136,2026-07-13 00:00:00-04:00,78.050003,83.860001,77.279999,83.300003,61577,9.590844
1053,2026-03-12 00:00:00-04:00,93.739998,101.800003,93.699997,100.459999,92219,9.219391
51,2022-03-17 00:00:00-04:00,98.099998,107.570000,97.879997,106.639999,15091,8.794127
1049,2026-03-06 00:00:00-05:00,84.839996,94.550003,83.169998,92.690002,197523,8.523590
1148,2026-07-29 00:00:00-04:00,84.949997,91.050003,84.949997,90.739998,22489,7.908196
1068,2026-04-02 00:00:00-04:00,100.300003,109.709999,99.019997,109.029999,0,7.779750
40,2022-03-02 00:00:00-05:00,107.540001,115.050003,106.489998,112.930000,59273,7.583118
1045,2026-03-02 00:00:00-05:00,77.000000,82.110001,75.790001,77.739998,143671,7.257166
53,2022-03-21 00:00:00-04:00,107.370003,116.730003,107.089996,115.620003,11620,7.124991
1144,2026-07-23 00:00:00-04:00,95.239998,102.010002,94.879997,100.690002,48567,7.037316


In [15]:
#Analisando os 10 dias com Maiores Quedas

maiores_quedas = df.nsmallest(10, ['Variacao_Diaria_%'])
display(maiores_quedas)

,Data,Valor_Abertura,Valor_Maximo,Valor_Minimo,Fechamento,Volume,Variacao_Diaria_%
1067,2026-04-01 00:00:00-04:00,103.900002,105.900002,98.459999,101.160004,61629,-14.524711
1071,2026-04-08 00:00:00-04:00,103.400002,104.000000,90.010002,94.750000,82140,-13.288183
45,2022-03-09 00:00:00-05:00,129.570007,131.630005,108.000000,111.139999,42740,-13.158309
1051,2026-03-10 00:00:00-04:00,92.449997,95.080002,81.160004,87.800003,144614,-11.277280
1060,2026-03-23 00:00:00-04:00,114.900002,115.099998,96.250000,99.940002,113332,-10.918976
144,2022-08-01 00:00:00-04:00,103.870003,104.370003,99.099998,100.029999,25670,-9.071905
1078,2026-04-17 00:00:00-04:00,98.250000,98.970001,86.080002,90.379997,85233,-9.065301
1146,2026-07-27 00:00:00-04:00,93.889999,93.889999,87.449997,88.360001,34887,-8.700143
1189,2026-09-25 00:00:00-04:00,106.519997,100.260002,96.389999,97.440002,63901,-8.592867
125,2022-07-05 00:00:00-04:00,111.610001,114.690002,101.110001,102.769997,43356,-7.936935
